### PR TITLE
CAMEL-10952 Initial work adding Kinesis Firehose

### PR DESCRIPTION
**Pull request by tigertoes**

### CODE REVIEW

File: components/camel-aws/src/test/java/org/apache/camel/component/aws/firehose/KinesisFirehoseProducerTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        kinesisFirehoseProducer = new KinesisFirehoseProducer(kinesisFirehoseEndpoint);
+    }
+
+    @Test
+    public void shouldPutRecordIntoStreamWhenProcessingExchange() throws Exception {
+        kinesisFirehoseProducer.process(exchange);
+        verify(outMessage).setHeader(KinesisFirehoseConstants.RECORD_ID, RECORD_ID);
+    }
+
+}
```
Comment (oscerd) [inline components/camel-aws/src/test/java/org/apache/camel/component/aws/firehose/KinesisFirehoseProducerTest.java]: Can you please run mvn -Psourcecheck for the codestyle and fix errors?

Thanks!

**Comment by oscerd** [comment]
Looks promising! Thanks!

@davsclaus can you please review when you get a chance?

File: components/camel-aws/src/main/java/org/apache/camel/component/aws/firehose/KinesisFirehoseEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.component.aws.kinesis.KinesisConsumer;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+
+@UriEndpoint(firstVersion = "2.19.0", scheme = "aws-kinesis-firehose", title = "AWS Kinesis Firehose", syntax = "aws-kinesis-firehose:streamName",
```
Comment (davsclaus) [inline components/camel-aws/src/main/java/org/apache/camel/component/aws/firehose/KinesisFirehoseEndpoint.java]: You need to set producerOnly = true here when we can only do producer

**Comment by davsclaus** [comment]
It may be nice to be able to set that awsClient option on the component so all endpoints uses that, so you dont have to refer to it in the endpoint uri, with the # style.

And if its on component level we can make it easier to configure using spring boot in its application.properties file.

File: components/camel-aws/src/test/java/org/apache/camel/component/aws/firehose/KinesisFirehoseProducerTest.java (hunk 1 above)
Comment (tigertoes) [inline components/camel-aws/src/test/java/org/apache/camel/component/aws/firehose/KinesisFirehoseProducerTest.java]: I've run this locally from the `components/camel-aws/` directory and I don't see any errors in output. Earlier commits had style errors that I had corrected prior to submitting to this PR.

**Comment by oscerd** [comment]
@tigertoes don't forget to add the component into the export section of pom here: https://github.com/apache/camel/blob/master/components/camel-aws/pom.xml#L33-L47

File: components/camel-aws/src/test/java/org/apache/camel/component/aws/firehose/KinesisFirehoseProducerTest.java (hunk 1 above)
Comment (oscerd) [inline components/camel-aws/src/test/java/org/apache/camel/component/aws/firehose/KinesisFirehoseProducerTest.java]: Great! Thanks! :-)

I usually remember to contributors about the code style

**Comment by tigertoes** [comment]
I've noticed that after building locally the build process has modified files in META-INF of camel-aws-starter. Do these changes also need to be committed in?

**Comment by davsclaus** [comment]
Yeah or if you dont include them they will be re-generated automatic. 

@oscerd looks good you are welcome to take this and get it merged

**Comment by oscerd** [comment]
Sure. I'll merge tomorrow

**Comment by oscerd** [comment]
Thanks. The PR has been merged. Can you close this PR?

**Comment by tigertoes** [comment]
Thanks for all the feedback and merging!

**Comment by oscerd** [comment]
you're welcome! :-)

**Pull request closed without merging**